### PR TITLE
lnwallet+walletrpc: list leases and parameterize duration

### LINKED ISSUE #4709: [Feature] Ability to set expiration on LeaseOutput
### Background

I dug into it and it looks like the `LeaseOutput` API falls back to using `btcwallet`'s default leasing time of [10 minutes](https://github.com/btcsuite/btcwallet/blob/a7f551a6301c6368084a522ca67009981b3dc0b6/wtxmgr/tx.go#L29). 

I realize I can keep calling it to reset the timer but the ability to programmatically set it to a certain long running time without repeated calls over and over again would be useful. Happy to take a stab at it, but wanted to open the ticket first for discussion. 

Looks like it'll have to be passed all the way through to: 

https://github.com/btcsuite/btcwallet/blob/a7f551a6301c6368084a522ca67009981b3dc0b6/wtxmgr/tx.go#L1156-L1176

### PR DESCRIPTION
**Pull request by joostjager**
Allows settings arbitrary lock durations. Also adds a `ListLeases` method to get an overview of all current leases. This call is particularly useful to not lose track of leases when longer durations are used.

Depends on https://github.com/btcsuite/btcwallet/pull/736

Fixes #4709

### CODE REVIEW

File: lnrpc/walletrpc/psbt.go [hunk 1]
```diff
@@ -55,26 +47,26 @@ func verifyInputsUnspent(inputs []*wire.TxIn, utxos []*lnwallet.Utxo) error {
...
+func lockInputs(w lnwallet.WalletController, packet *psbt.Packet) (
+	[]*wtxmgr.LockedOutput, error) {
 
-	locks := make([]*utxoLock, len(packet.UnsignedTx.TxIn))
+	locks := make([]*wtxmgr.LockedOutput, len(packet.UnsignedTx.TxIn))
 	for idx, rawInput := range packet.UnsignedTx.TxIn {
-		lock := &utxoLock{
-			lockID:   LndInternalLockID,
-			outpoint: rawInput.PreviousOutPoint,
+		lock := &wtxmgr.LockedOutput{
```
Comment (guggero) [inline lnrpc/walletrpc/psbt.go:55]: :+1:

**Review by guggero — APPROVED** [review]
Very useful, thank you! LGTM :100:

**Comment by Roasbeef** [comment]
Needs a rebase now that the dependent PR has been merged. We'll likely also want to update all the module versions as well before this PR lands.

**Comment by joostjager** [comment]
Rebased

**Review by wpaulino — APPROVED** [review]

**Comment by joostjager** [comment]
Anything blocking updating the `btcwallet` module versions?

**Comment by wpaulino** [comment]
We have other upcoming work in btcwallet we plan to get in soon as well, so we might not need to tag yet.

**Pull request merged**